ts-rs/src/lib.rs

```diff
@@ -289,18 +288,25 @@ pub trait TS {
 
     /// Declaration of this type, e.g. `interface User { user_id: number, ... }`.
     /// This function will panic if the type has no declaration.
+    ///
+    /// If this type is generic, then all provided generic parameters will be swapped for
+    /// placeholders, resulting in a generic typescript definition.
+    /// Both `SomeType::<i32>::decl()` and `SomeType::<String>::decl()` will therefore result in
+    /// the same TypeScript declaration `type SomeType<A> = ...`.
     fn decl() -> String {
         panic!("{} cannot be declared", Self::name());
     }
 
+    /// Declaration of this type using the supplied generic arguments.
+    /// The resulting TypeScript definition will not be generic. For that, see `TS::decl()`.
+    /// If this type is not generic, then this function is equivalent to `TS::decl()`.
+    fn decl_concrete() -> String {
+        panic!("{} cannot be declared", Self::name());
+    }
+
```

Comment:
Here's the user-visible effect of this change.
For
```rust
struct Generic<A, B> { a: A, b: B }
```,
`Generic::<i32, String>::decl_concrete()` returns `type Generic = { a: i32, b: String };`. 
`Generic::<i32, String>::decl()` swaps out the generic parameters `i32` and `String` out for on-the-fly generated dummy types `A` and `B`, resulting in `type Generic<A, B> = { a: A, b: B };`